staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -234,6 +236,14 @@ func (s *EtcdOptions) addEtcdHealthEndpoint(c *server.Config) error {
 		return healthCheck()
 	}))
 
+	readyCheck, err := storagefactory.CreateReadyCheck(s.StorageConfig, c.DrainedNotify())
```

Comment:
I was wondering if we shouldn't use a different channel here, although given it's actually used as a stopCh it actually makes sense.